### ISSUE TITLE
It takes a long time to install the chaincode

### ISSUE THREAD

**Issue by VolcanoWCAN** [kehm/ehrblockchain#5]
I learned a lot from this project. Thank you for your excllent work！
I ran this project successfully before, but today I encountered another problem.
So I came to you again for help
This time， I still run it based on your project and haven't changed the code.
When I run ./start.sh, then it stops and I wait for a long time without success.
This situation has happened before. Then I just restart the project.
However this time it does not help！

The log outpus like this：
Creating docker containers:Start
Creating volume "ehrnetwork_peer0.pharmacy1.example.com" with default driver
Creating volume "ehrnetwork_peer0.practitioner1.example.com" with default driver
Creating volume "ehrnetwork_ca.pharmacy1.example.com" with default driver
Creating volume "ehrnetwork_ca.hospital1.example.com" with default driver
Creating volume "ehrnetwork_ca.practitioner1.example.com" with default driver
Creating volume "ehrnetwork_orderer0.practitioner1.example.com" with default driver
Creating volume "ehrnetwork_peer1.hospital1.example.com" with default driver
Creating volume "ehrnetwork_orderer0.pharmacy1.example.com" with default driver
Creating volume "ehrnetwork_peer1.practitioner1.example.com" with default driver
Creating volume "ehrnetwork_peer1.pharmacy1.example.com" with default driver
Creating volume "ehrnetwork_orderer0.hospital1.example.com" with default driver
Creating volume "ehrnetwork_peer0.hospital1.example.com" with default driver
Creating orderer0.pharmacy1.example.com ... done
Creating couchdb4 ... done
Creating ca.hospital1.example.com ... done
Creating orderer0.practitioner1.example.com ... done
Creating couchdb0 ... done
Creating couchdb5 ... done
Creating ca.practitioner1.example.com ... done
Creating orderer0.hospital1.example.com ... done
Creating couchdb3 ... done
Creating ca.pharmacy1.example.com ... done
Creating couchdb2 ... done
Creating couchdb1 ... done
Creating peer1.practitioner1.example.com ... done
Creating peer1.pharmacy1.example.com ... done
Creating peer0.hospital1.example.com ... done
Creating peer1.hospital1.example.com ... done
Creating peer0.practitioner1.example.com ... done
Creating peer0.pharmacy1.example.com ... done
Creating cli0 ... done
Creating docker containers:Done
Waiting 10s for orderers to get ready
Send create channel request:Start
Send create channel request:Done
Adding peers to channel:Start
Adding peers to channel:Done
Define anchor peers:Start
Define anchor peers:Done
Install chaincode:Start

And the program stuck here
It is useless to wait for a long time
What should I do to solve the problem？
Thank you very much！！！

**Comment by VolcanoWCAN** [comment]
I learned a lot from this project. Thank you for your excllent work！
I ran this project successfully before, but today I encountered another problem.
So I came to you again for help
This time， I still run it based on your project and haven't changed the code.
When I run ./start.sh, then it stops and I wait for a long time without success.
This situation has happened before. Then I  just restart the project.
However this time it does not help！

The log outpus like this：
Creating docker containers:Start
Creating volume "ehrnetwork_peer0.pharmacy1.example.com" with default driver
Creating volume "ehrnetwork_peer0.practitioner1.example.com" with default driver
Creating volume "ehrnetwork_ca.pharmacy1.example.com" with default driver
Creating volume "ehrnetwork_ca.hospital1.example.com" with default driver
Creating volume "ehrnetwork_ca.practitioner1.example.com" with default driver
Creating volume "ehrnetwork_orderer0.practitioner1.example.com" with default driver
Creating volume "ehrnetwork_peer1.hospital1.example.com" with default driver
Creating volume "ehrnetwork_orderer0.pharmacy1.example.com" with default driver
Creating volume "ehrnetwork_peer1.practitioner1.example.com" with default driver
Creating volume "ehrnetwork_peer1.pharmacy1.example.com" with default driver
Creating volume "ehrnetwork_orderer0.hospital1.example.com" with default driver
Creating volume "ehrnetwork_peer0.hospital1.example.com" with default driver
Creating orderer0.pharmacy1.example.com     ... done
Creating couchdb4                           ... done
Creating ca.hospital1.example.com           ... done
Creating orderer0.practitioner1.example.com ... done
Creating couchdb0                           ... done
Creating couchdb5                           ... done
Creating ca.practitioner1.example.com       ... done
Creating orderer0.hospital1.example.com     ... done
Creating couchdb3                           ... done
Creating ca.pharmacy1.example.com           ... done
Creating couchdb2                           ... done
Creating couchdb1                           ... done
Creating peer1.practitioner1.example.com    ... done
Creating peer1.pharmacy1.example.com        ... done
Creating peer0.hospital1.example.com        ... done
Creating peer1.hospital1.example.com        ... done
Creating peer0.practitioner1.example.com    ... done
Creating peer0.pharmacy1.example.com        ... done
Creating cli0                               ... done
Creating docker containers:Done
Waiting 10s for orderers to get ready
Send create channel request:Start
Send create channel request:Done
Adding peers to channel:Start
Adding peers to channel:Done
Define anchor peers:Start
Define anchor peers:Done
Install chaincode:Start

****And the program stuck here**
 It is useless to wait for a long time**
What should I do to solve the problem？
Thank you very much！！！

**Comment by kehm** [comment]
Hi there. I'm glad you find the project interesting.
I would suggest that instead of running start.sh you run every line in the script one by one in your command line. This way you will get the complete output from each function call which makes it easier to debug. 
In other words, copy every line from 9-34 and run them one by one in your CLI, and remember to remove "&> /dev/null" from the "docker exec" calls so that you get the output.

**Comment by VolcanoWCAN** [comment]
Thank you!
I will try this method later. And I think it's a good method that can help me to understand this project！

I learned a lot from your project, and I also tried to solve the problems that came up by myself. It gave me a lot of inspiration and helped me better understand the entire architecture and application method of the blockchain. I also encountered many questions in it, thanks to your patience!

**Comment by VolcanoWCAN** [comment]
Hi, Kehm:


Here, I have some other questions about this project. And it's about the java application side:
&nbsp; &nbsp; &nbsp; &nbsp; 1、When the program is running, we need to log in. In this case, I enter "admin" in the login window, and it will prompt the login failure. Then I used "admin" again, I found that it can log in successfully. I found that no password was used during this process. And a folder called "user " occurs. There are some information in this folder. I want to know how this is done for user authentication? What is the reason for this?
&nbsp; &nbsp; &nbsp; &nbsp; 2、
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; I had add mappings to /etc/hosts.
&nbsp; &nbsp; &nbsp; &nbsp; When I enroll the application，it outputs that：



23.02.13: ERROR: An enrollment error occured

23.02.28: Enrolled as 'admin' with 'hospital1.surgery'

23.02.30: Channel 'providerschannel'. Length: 5

23.02.40: Assembling transaction...

23.02.40: Transaction failed


&nbsp; &nbsp; &nbsp; &nbsp; The log outputs：
&nbsp; &nbsp; &nbsp; &nbsp; DEBUG ServiceDiscovery - Channel providerschannel peer Peer{ id: 2, name: peer0.hospital1.example.com, channelName: providerschannel, url: grpcs://172.18.0.40:7051} completed chaincode query request
&nbsp; &nbsp; &nbsp; &nbsp; ERROR ServiceDiscovery - Error failed constructing descriptor for chaincodes:<name:"RecordRelationshipContract" &gt;
WARN ServiceDiscovery - Channel providerschannel failed to find all layouts for chaincodes. Expected: 1 and found: 0


&nbsp; &nbsp; &nbsp; &nbsp; I learned a lot from your project, and I also tried to solve the problems that came up by myself.&nbsp;
&nbsp; &nbsp; &nbsp; &nbsp; It gave me a lot of inspiration and helped me better understand the entire architecture and application method of the blockchain. I also encountered many questions in it, thanks to your patience!



------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Kjell-Erik Marstein"<notifications@github.com&gt;;
发送时间:&nbsp;2020年2月20日(星期四) 晚上8:46
收件人:&nbsp;"kehm/ehrblockchain"<ehrblockchain@noreply.github.com&gt;;
抄送:&nbsp;"王灿"<543711843@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [kehm/ehrblockchain] It takes a lon (#5)




Hi there. I'm glad you find the project interesting.
 I would suggest that instead of running start.sh you run every line in the script one by one in your command line. This way you will get the complete output from each function call which makes it easier to debug.
 In other words, copy every line from 9-34 and run them one by one in your CLI, and remember to remove "&amp;&gt; /dev/null" from the "docker exec" calls so that you get the output.

**Comment by VolcanoWCAN** [comment]
Again it stops here！
I  had run it successfully.  But now it cannot work well again.
![2020-02-21 00-24-43屏幕截图](https://user-images.githubusercontent.com/33406143/74956338-e97aca00-5440-11ea-8734-1fac1ec18918.png)

**Comment by VolcanoWCAN** [comment]
I run the scripts one by one.
It stops at this place.
It stops at Instantiate chaincode on channel！
But I dont know how to deal with it!
![2020-02-20 23-39-01屏幕截图](https://user-images.githubusercontent.com/33406143/74956335-e8e23380-5440-11ea-9f27-180403211984.png)

**Comment by VolcanoWCAN** [comment]
How can I solve this problem？
Sometimes I wait for a long time and then it can work.
This time I run the scripts one by one, but it stuck here!!!!
And I don't know why.

**Comment by kehm** [comment]
I would check the Docker containers to try and identify where the error is. I would assume that the chaincode is not properly installed or that the specified orderer is not available.
Try "docker ps" to list all running docker containers and then "docker logs [container]" each of them to check for error messages.

**Comment by VolcanoWCAN** [comment]
Thank you！ I have sovled all the problems！
